certbot/certbot/_internal/tests/main_test.py

```diff
@@ -621,6 +621,46 @@ def test_update_configurator(self):
         new_config = self._call('--cert-name example.com --apache'.split())
         assert new_config['renewalparams']['authenticator'] == 'apache'
 
+    def test_only_intended_changes(self):
+        """ Check that we don't accidentally modify anything that we didn't mean to """
+        named_mock = mock.Mock()
+        named_mock.name = 'apache'
+
+        self.mocks['pick_installer'].return_value = named_mock
+        self.mocks['pick_auth'].return_value = named_mock
+        self.mocks['find_init'].return_value = named_mock
+
+        new_config = self._call('--cert-name example.com --apache'.split())
+        # Undo the changes we made in calling and in testing
+        new_config['renewalparams']['authenticator'] = 'nginx'
+        new_config['renewalparams']['installer'] = 'nginx'
+        del new_config['renewalparams']['config_dir']
+        new_config['version'] = self.original_config['version']
+
+        assert new_config == self.original_config
+
+    @mock.patch('certbot._internal.hooks.validate_hooks')
+    def test_staging_used(self, unused_validate_hooks):
+        """ Check that we use the staging server for the dry run """
+        assert self.original_config['renewalparams']['server'] == \
+            'https://acme-v02.api.letsencrypt.org/directory'
+
+        self._call('--cert-name example.com --pre-hook'.split() + ['echo pre'])
+
+        assert 'staging' in self.mocks['_init_le_client'].call_args.args[0].server
+        assert 'staging' in self.mocks['_get_and_save_cert'].call_args.args[1].server
+
+    def test_account_updates(self):
+        """ Even though we can't run the dry run with the new account id, it should still
+            update in the renewal conf file.
+        """
```

Comment:
So...`reconstitute` won't load the account into the config file if server is set, because of that magic `set_by_user` logic that considers account to have implicitly been set if server was set. My impulse here is instead of checking to see if the values changed, just check if either was `set_by_user` and fail based on that.

Reply:
This is only a problem for `--server`, not `--account`, so we could technically do the change check for just `account`, but that seems even less likely to be the thing someone's trying to change.

Reply:
lol, of course our integration tests do exactly that.

Reply:
ok, if server was set by user and is the same as the one in the file, we can just manually load in the account from the saved config file, I guess.

Reply:
This is a response to the thread current ending at https://github.com/certbot/certbot/pull/9870/files/710e17c69f048a1d0ff53f7f935993db03d17640..07a35a8c8f370cb1ff163f948eb550c9075bb8dc#r1462604621 which I think GitHub isn't always clear about when threads are continued as part of another review.

> It gets at a more fundamental question of if we're ok helping people make untested changes or not. If someone really wants to make the change without testing though, I suppose they can always just go in and edit the configuration file. I think I'm leaning towards the error out and provide the certonly suggestion though, what do you think?

My opinion is that we should avoid helping people make untested changes to their renewal config files. Like you said, we can't completely stop them, but I think it's a bad idea and I wouldn't do it myself. Who wants to deal with renewal failures weeks later when you may no longer even remember what was changed?

I think erroring out is good. I like the suggestion of `certbot renew` slightly better than `certonly` because `certonly` blindly overwrites the previous renewal parameters with the settings used to invoke Certbot (with minor exceptions about things like changing the key type) while `certbot renew` merges the current settings with the old ones like `certbot reconfigure` does. I think this difference makes `certbot certonly` a bit more cumbersome because you have to respecify all values and any you forget will be dropped. I don't have that strong of a preference though so if you still prefer `certonly` despite this, this is fine with me.

> So...`reconstitute` won't load the account into the config file if server is set, because of that magic `set_by_user` logic that considers account to have implicitly been set if server was set.

Very nice catch! I missed that we were dropping the account from the config file if server was set during my testing.

Reply:
That's a good point, I'll switch to renew. Even if nothing else, we should match what our online instructions say.